### PR TITLE
added `sign` feature (optional) in `is.sorted`

### PR DESCRIPTION
**Pull request by ironmaniiith**
Refer #220

### CODE REVIEW

**Comment by ironmaniiith** [comment]
Will update README in later PR

File: test/test.js [hunk 1]
```diff
@@ -3128,52 +3128,96 @@ describe('object checks', function() {
...
+            expect(is.not.sorted(arr1, '<')).to.be.true;
+
+            var arr2 = [5, 4, 4, 3, 1];
+            // default behaviour:
+            expect(is.not.sorted(arr2)).to.be.true;
+            // extended behaviour:
+            expect(is.not.sorted(arr2, '>=')).to.be.true;
+            expect(is.not.sorted(arr2, '>')).to.be.true;
+            expect(is.not.sorted(arr2, '<=')).to.be.false;
+            expect(is.not.sorted(arr2, '<')).to.be.true;
```
Comment (jdalton) [inline test/test.js]: Duplicating tests for `not` and `any` seems like overkill. We should have tests that ensure `not` and `any` do their thing and then it's safe to assume the behavior applied to others. What we can do is instead ensure the `sorted` method exists on the `not` and `any` object which is something to check for.

Comment (ironmaniiith) [inline test/test.js]: very `true`, sometimes it becomes irritating seeing the duplicate tests. Handled in latest commit

**Comment by jdalton** [comment]
I think this needs another cleanup pass.

**Comment by ironmaniiith** [comment]
@jdalton, in test file or the code.?

**Comment by jdalton** [comment]
In the test file. You can see there are a couple of orphan comments.

**Comment by ironmaniiith** [comment]
Ok, I just saw there were absolutely no comments in test file. Anyways there was no feature like this earlier as well (default behaviour and extended behaviour) hence I kept those comments.

File: test/test.js [hunk 2]
```diff
@@ -3128,31 +3128,50 @@ describe('object checks', function() {
...
             var arr = [1, 2, 3, 5, 4];
             expect(is.not.sorted(arr)).to.be.true;
+            expect(is.not.sorted(arr, '>')).to.be.true;
         });
     });
     describe('is.all.sorted', function() {
         it('should return true if all given arrays are sorted', function() {
             var arr1 = [1, 2, 3, 4, 5];
             var arr2 = [5, 6, 7, 8, 9];
-            var arr3 = [10];
```
Comment (jdalton) [inline test/test.js]: Shouldn't the single element case `[10]` be covered `is.sorted` test for `'should return true if given array is sorted'`?

File: test/test.js [hunk 3]
```diff
@@ -3196,14 +3186,12 @@ describe('array checks', function() {
             var arr2 = [10, 6, 7, 8, 9];
             expect(is.any.sorted(arr1, arr2)).to.be.true;
             expect(is.any.sorted([arr1, arr2])).to.be.true;
-            // extended behaviour not available with 'any' interface
```
Comment (jdalton) [inline test/test.js]: ^ These were the orphan comments. See how they don't pertain to any code as they're at the bottom of their code block.

File: test/test.js [hunk 4]
```diff
@@ -3160,17 +3156,13 @@ describe('array checks', function() {
     describe('is.not.sorted', function() {
         it('should return false if given array is sorted', function() {
             var arr = [1, 2, 3, 4, 5];
-            // default behaviour:
```
Comment (jdalton) [inline test/test.js]: This is not an orphan comment as it has code it relates to right below it.

File: test/test.js [hunk 5]
```diff
@@ -3196,14 +3186,12 @@ describe('array checks', function() {
...
             expect(is.any.sorted([arr1, arr2])).to.be.true;
-            // extended behaviour not available with 'any' interface
         });
         it('should return false if all given arrays are not sorted', function() {
             var arr1 = [6, 2, 3, 4, 5];
             var arr2 = [9, 5, 6, 7, 8];
             expect(is.any.sorted(arr1, arr2)).to.be.false;
             expect(is.any.sorted([arr1, arr2])).to.be.false;
-            // extended behaviour not available with 'any' interface
         });
```
Comment (jdalton) [inline test/test.js]: ^ This is orphan...

File: test/test.js (hunk 2 above)
Comment (ironmaniiith) [inline test/test.js]: I felt it like very obvious hence I removed it. If you feel the need for it, I'll add.

Comment (jdalton) [inline test/test.js]: Doesn't the single case cover the the whole `length -1` concern?

Comment (jdalton) [inline test/test.js]: My point before was that there is no reason for every method to have `not` and `any` tests. Instead we should test that the `not` and `any` helpers do their job generically and then just test for the existence of the `not.sorted` and `any.sorted` methods.

Comment (ironmaniiith) [inline test/test.js]: Not the problem with this implementation.

Comment (jdalton) [inline test/test.js]: It still could be. Tests help prevent regressions.

Comment (ironmaniiith) [inline test/test.js]: Added

File: test/test.js (hunk 3 above)
Comment (ironmaniiith) [inline test/test.js]: Ok, so which one is best option.
* Keep that comment, but at the top
* Remove that comment
* Something else :stuck_out_tongue:

**Comment by jdalton** [comment]
You're missing the point. I'll merge and then show you.

**Comment by ironmaniiith** [comment]
Best option :stuck_out_tongue: , I'll learn from that and will take care later on

**Pull request merged**